server/app/services/rpc_server.rb

```diff
@@ -93,15 +116,19 @@ def handling_actor(grid_id, name)
     @handlers[grid_id] ||= {}
     unless @handlers[grid_id][name]
       grid = Grid.find(grid_id)
-      @handlers[grid_id][name] = HANDLERS[name].new(grid) if grid
+      if grid
+        @handlers[grid_id][name] = HANDLERS[name].new(grid)
+      end
     end
 
     @handlers[grid_id][name]
   end
 
   # @param [Faye::Websocket] ws
-  # @param [Array] message
+  # @param [Object] message
   def send_message(ws, message)
-    ws.send(MessagePack.dump(message).bytes)
+    EM.next_tick {
+      ws.send(MessagePack.dump(message.as_json).bytes)
```

Comment:
Confused by the `message.as_json`... the agent isn't expecting a JSON-encoded string, it should just be the binary `MessagePack.dump` of the actual ruby hash from the `RpcSerializer.to_hash`? What does this `.as_json` do?